fragalysis_api/xcimporter/conversion_pdb_mol.py

```diff
@@ -231,14 +231,14 @@ def create_metadata_file(self, directory, file_base, smiles_file=None):
         Metadata .csv file prepared for each ligand
 
         params: file_base and smiles
-        returns: metadata.csv and smiles.txt file for each ligand
+        returns: .mol file for the ligand
```

Comment:
this description has reverted back to what it was before - is that correct?

Reply:
In the bit where you're trying to determine chain and crystal names by string manipulation, you're going to have lots of issues if people include underscores in their crystal names. This is another reason to start with a single metadata file. I would do the following:

1. When the API is first called, before any monomerization or alignment, create an initial metadata file using the filename e.g. Warren.pdb as the crystal name - this becomes crystal_name = Warren. 

```
crystal_name, fragalysis_label
Warren, Warren
```

2. If the monomerization step is run for each crystal, use its old crystal name as crystal name, and duplicate row for each file that is split out, so if Warren.pdb is split out into chain A and B, the metadata file contains:

```
crystal_name, fragalysis_label
Warren, Warren_A
Warren, Warren_B
```

3. When the ligands are split out, we then duplicate those rows for each crystal, so if Warren_A has 3 ligands and Warren_B has 1:

```
crystal_name, fragalysis_label
Warren, Warren_0A
Warren, Warren_1A
Warren, Warren_2A
Warren, Warren_3B
```

You can achieve this all with one function, probably:

```
def rename_for_fragalysis(rename_dict, metadata_file):
    metadata_frame = pandas.read_csv(metadata_file)
    for old_fragalysis_label in rename_dict.keys():
        if len(rename_dict[old_fragalysis_label]==1:
            # just change the fragalysis_label value in the relevant file here
        else:
             # duplicate the row len(rename_dict[old_fragalysis_label]) times
             # changing the fragalysis_label value each time
    
    # write out the metadata.csv file with the new data in it
    return whatever

```

where `metadata_file` contains the csv file you're working on and `rename_dict` is a dictionary where each key is the old value of the fragalysis_label value that you want to change, and the value for each key is a list of the new fragalysis_labels that correspond to it. e.g. to go from step 2 to 3 above:

```
rename_dict = {'Warren_A': ['Warren_0A', 'Warren_1A', 'Warren_2A'],
                           'Warren_B': ['Warren_3B']}

metadata_file = 'metadata.csv'

rename_for_fragalysis(rename_dict, metadata_file)